fdk/customer_code.py

```diff
@@ -0,0 +1,38 @@
+# All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+from importlib import util
+
+
+class Function(object):
+
+    def __init__(self, func_module_path, entrypoint="handler"):
+        self.spec, self.mod = self.import_from_source(func_module_path)
+        self._executed = False
+        self._entrypoint = entrypoint
+
+    @staticmethod
+    def import_from_source(func_module_path):
+        func_module_spec = util.spec_from_file_location(
+            "func", func_module_path
+        )
+        func_module = util.module_from_spec(func_module_spec)
+        return func_module_spec, func_module
+
+    def handler(self):
+        if self._executed is False:
+            self.spec.loader.exec_module(self.mod)
+            self._executed = True
+
+        return getattr(self.mod, self._entrypoint)
```

Comment:
`getattr()` can raise an exception of the attribute it's looking up doesn't exist.  Do we have any edge cases where where a user creates a `Function()` but overrides `_entrypoint` with a name of a function that we can't find?  This is a long-winded way of asking if we need any error handling here to make that problem more visible to the user, assuming I've understood what's going on here correctly anyway.

Reply:
User's don't have to use class `Function` and instances of that at all, it's not their call.
class `Function` is only for internal use.